Sukhorukov/app.js

```diff
@@ -0,0 +1,65 @@
+const readline = require('readline-sync');
+const fs = require('fs');
+
+class Game {
+    constructor(logger, logFile){
+        this.logger = logger;
+        this.logFile = logFile;
+        this.play();
+    }
+
+    play(){
+        while (this.answer !== 0){
+            this.question();
+            if(!this.isValid()){
+                this.question();
+            }
+            if(this.isWin()){
+                this.logger.log(this.logFile, '1');
+                console.log('Вы выиграли');
+            } else {
+                if(this.answer === 0)
+                    break;
+                this.logger.log(this.logFile, '0');
+                console.log('Вы проиграли');
+            }
+        }
+    }
+
+    question(){
+        let answer = readline.question('Enter 1(head), 2(tail), 0(exit)');
+        this.answer = +answer
+    }
+
+    flipCoin() {
+        return Math.floor(Math.random() * 2 + 1);
+    }
+
+    isWin() {
+        return this.flipCoin() === this.answer;
+    }
+
+    isValid() {
+        return this.answer === 1
```

Comment:
В таком виде выглядит более расширяемым :)
```suggestion
        return [0, 1, 2].includes(this.answer);
```